app/src/main/java/com/duckduckgo/app/browser/orientation/JsOrientationHandler.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.orientation
+
+import android.content.pm.ActivityInfo
+import androidx.fragment.app.FragmentActivity
+import com.duckduckgo.app.global.view.requestJsOrientationChange
+import com.duckduckgo.js.messaging.api.JsCallbackData
+import org.json.JSONObject
+
+class JsOrientationHandler {
+
+    /**
+     * Updates the activity's orientation based on provided JS data
+     *
+     * @return response data
+     */
+    fun updateOrientation(data: JsCallbackData, activity: FragmentActivity?): JsCallbackData {
```

Comment:
Rather than passing the activity you could pass the result of `activity.requestJsOrientationChange(matchedOrientation)` which is safer.

Reply:
Thanks, this is one of those parts I was iffy about, but couldn't come up with anything better 😅  Could you expand your suggestion a bit more? This method determines what value we should pass to `requestJsOrientationChange` and determines error cases as it goes, so I am not sure how we can pass it the result of `requestJsOrientationChange`? Alternatively, we could have `requestJsOrientationChange` handle different error cases and return appropriate codes, but that doesn't seem right either...

Reply:
It feels the only intention to `JsOrientationHandler` class to extract this code our of the `BrowserTabFragment`. But otherwise it's an implementation detail of `BrowserTabFragment`, always to be instantiated by it not to leak any memory.

However having `JsOrientationHandler` as a separate class, in a separate package that accepts a generic activity as a parameter is the wrong contract, for the same above reason and because it sets easy precedent for others to copy-paste anti-patterns.

To avoid leaking what I would do is:
* Move `JsOrientationHandler` inside the `BrowserTabFragment.kt` file as a private class
* (forget about my comment about using DI, ie. instantiate as `private val jsOrientationHandler = JsOrientationHandler()` and leave a comment as to why
* change the contract of `updateOrientation` to be
```kotlin
fun updateOrientation(data: JsCallbackData, browserTabFragment: BrowserTabFragment)
```

Reply:
It's true - the intention of `JsOrientationHandler` was to keep that code outside of `BrowserTabFragment`. We could just plop it back in as you suggest, Aitor. 

Another idea I've been thinking about is creating a `FragmentActivity` extension inside the `com.duckduckgo.app.browser` package so it's not as exposed and move the logic of `updateOrientation` to there.

What do you guys think? I have no strong opinions here.

Reply:
My main point is that this is an implementation detail of `BrowserTabFragment`, not of any other generic type. So we should not create public API surface on generic types when we can't/should not use them

Reply:
Makes sense - addressed